src/components/Modal/Modal.js

```diff
@@ -137,17 +137,19 @@ class Modal extends Component<Props, State> {
   };
   handleClose = (event: MouseOrKeyboardEvent) => {
     const { onClose } = this.props;
-    const { isFullscreen } = this.state;
+    const { isFullscreen, isClosing } = this.state;
 
-    this.setState({ isClosing: true });
+    if (!isClosing) {
```

Comment:
It may be nicer to use a guard clause here, such as:
```
if (isClosing) {
  return;
}
```
rather than enclosing the entire logic within the if block?